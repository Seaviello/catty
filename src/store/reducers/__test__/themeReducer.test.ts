import reducer, {DEFAULT_COLOR} from '../themeReducer';
import {
  RESET_COLOR,
  SET_COLOR,
  ThemeActionTypes,
} from '../../actions/themeActionTypes';

describe('themeReducer', () => {
  const initialState = reducer(undefined, {} as ThemeActionTypes);
  it('initial state should be properly initialized ', () => {
    expect(initialState).toEqual({
      appBackgroundColor: DEFAULT_COLOR,
    });
  });
  it('should handle setColor action', () => {
    const state = reducer(initialState, {
      type: SET_COLOR,
      payload: {color: 'red'},
    });
    expect(state).toMatchSnapshot();
  });
  it('should handle resetColor action', () => {
    const state = reducer(
      {...initialState, appBackgroundColor: 'red'},
      {type: RESET_COLOR},
    );
    expect(state).toMatchSnapshot();
  });
});
