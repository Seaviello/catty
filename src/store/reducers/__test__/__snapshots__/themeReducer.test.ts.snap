// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themeReducer should handle resetColor action 1`] = `
Object {
  "appBackgroundColor": "#90caf9",
}
`;

exports[`themeReducer should handle setColor action 1`] = `
Object {
  "appBackgroundColor": "red",
}
`;
